pages/data.py

```diff
@@ -141,12 +160,36 @@ def update_sub_tab(tab, store_demographics):
         df = df.drop(columns=[col for col in df.columns if col not in columns])
 
         return populate_datatable(df)
-      
+
+
+@callback(
+    Output('data-table', 'hidden_columns'), # Output hidden columns in the table
+    Output('button-clicked', 'children'), #updates button label
+    Input('button-clicked', 'n_clicks'), #number of clicks on the button
+    State('button-clicked', 'children') #State representing the current label of button
+)
+#Controls visibility of columns in trips table  and updates the label of button based on the number of clicks.
+def update_dropdowns_trips(n_clicks, button_label):
+    columns = perm_utils.get_allowed_trip_columns()
+    columns.update(
+        col['label'] for col in perm_utils.get_allowed_named_trip_columns()
+    )
+    if n_clicks % 2 == 0:
+        hidden_col = ["data.duration_seconds", "data.distance_meters","data.distance"]
+        button_label = 'Display columns with raw units'
+    else:
+        hidden_columns = ["data.duration", "data.distance_miles", "data.distance_km", "data.distance"]
+        hidden_col = [col for col in hidden_columns if col in columns]
```

Comment:
why do we need both `hidden_columns` and `hidden_col` in the `else` but not in the `if`
it seems like we should check, in both cases, whether the hidden columns exist